assembl/tests/views_tests/test_graphql.py

```diff
@@ -1938,3 +1938,19 @@ def __init__(self, filename, type):
     assert resources_center['headerImage'] is not None
     assert '/documents/' in resources_center['headerImage']['externalUrl']
     assert resources_center['headerImage']['title'] == 'new-img.png'
+
+def test_query_visits_analytics(discussion, graphql_request):
+    query = u"""
+query GetVisitsAnalytics {
+  visitsAnalytics {
+    sumVisitsLength
+    nbPageviews
+    nbUniqPageviews
+  }
+}
+"""
+    res = schema.execute(query, context_value=graphql_request)
+    assert res.data['visitsAnalytics'] is not None
+    assert res.data['visitsAnalytics']['sumVisitsLength'] == None
+    assert res.data['visitsAnalytics']['nbPageviews'] == None
+    assert res.data['visitsAnalytics']['nbUniqPageviews'] == None
```

Comment:
You should really add a test with non None values by mocking the result of `discussion.get_visits_time_series_analytic` with the `mock` library. There are some examples already in `test_api.py` and `test_view_auth.py`. We don't want to break things when we will upgrade to graphene 2 that changes the api of resolvers.

Reply:
And a test with a range to test start_date and end_date args.

Reply:
Yes but I don't know how to do it, as I said in https://github.com/assembl/assembl/pull/246#issuecomment-347901370

Reply:
In this particuliar case, I would do something like this:
```
    def test_query_visits_analytics(discussion, graphql_request):
        def fake_get_visits_time_series_analytics(self, start_date=None, end_date=None, only_fields=None):
             if "sum_visits_length" in only_fields:
                 return 15

        with mock.patch.object(models.Discussion, 'get_visits_time_series_analytics', fake_get_visits_time_series_analytics):
            
            ...do your graphql query...

Reply:
improvement done in https://github.com/assembl/assembl/pull/246/commits/295b190aeffffc3eff225453d6d338c5a28566d8